aptos-move/framework/aptos-framework/sources/configs/gas_schedule.move

```diff
@@ -1,88 +1,137 @@
+        GasSchedule {
+            entries: vector::empty(),
+        }
+    }
 
-        /// The flat minimum amount of gas required for any transaction.
-        /// Charged at the start of execution.
-        min_transaction_gas_units: u64,
+    public fun index_of(gas_schedule: &GasSchedule, key: &String): Option<u64> {
+        let len = vector::length(&gas_schedule.entries);
 
-        /// Any transaction over this size will be charged an additional amount per byte.
-        large_transaction_cutoff: u64,
+        let i = 0;
+        while (i < len) {
+            let entry = vector::borrow(&gas_schedule.entries, i);
+            if (&entry.key == key) {
+                return option::some(i)
+            };
+            i = i + 1;
+        };
 
-        /// The units of gas to be charged per byte over the `large_transaction_cutoff` in addition to
-        /// `min_transaction_gas_units` for transactions whose size exceeds `large_transaction_cutoff`.
-        intrinsic_gas_per_byte: u64,
+        option::none()
+    }
 
-        /// ~5 microseconds should equal one unit of computational gas. We bound the maximum
-        /// computational time of any given transaction at roughly 20 seconds. We want this number and
-        /// `MAX_PRICE_PER_GAS_UNIT` to always satisfy the inequality that
-        /// MAXIMUM_NUMBER_OF_GAS_UNITS * MAX_PRICE_PER_GAS_UNIT < min(u64::MAX, GasUnits<GasCarrier>::MAX)
-        /// NB: The bound is set quite high since custom scripts aren't allowed except from predefined
-        /// and vetted senders.
-        maximum_number_of_gas_units: u64,
+    public fun get(gas_schedule: &GasSchedule, key: &String): Option<u64> {
+        let idx_opt = index_of(gas_schedule, key);
 
-        /// The minimum gas price that a transaction can be submitted with.
-        min_price_per_gas_unit: u64,
+        if (option::is_none(&idx_opt)) {
+            return option::none()
+        };
 
-        /// The maximum gas unit price that a transaction can be submitted with.
-        max_price_per_gas_unit: u64,
+        let idx = option::destroy_some(idx_opt);
+        option::some(vector::borrow(&gas_schedule.entries, idx).val)
+    }
 
-        max_transaction_size_in_bytes: u64,
-        gas_unit_scaling_factor: u64,
-        default_account_size: u64,
+    public fun set(gas_schedule: &mut GasSchedule, key: String, val: u64): Option<u64> {
+        let idx_opt = index_of(gas_schedule, &key);
+
+        if (option::is_none(&idx_opt)) {
+            vector::push_back(&mut gas_schedule.entries, GasEntry { key, val });
+            option::none()
+        }
+        else {
+            let idx = option::destroy_some(idx_opt);
+            let entry = vector::borrow_mut(&mut gas_schedule.entries, idx);
+            let old = entry.val;
+            entry.val = val;
+            option::some(old)
+        }
+    }
+
+    public fun remove(gas_schedule: &mut GasSchedule, key: &String): Option<u64> {
```

Comment:
something doesn't quite feel right about these being exposed as public. I also didn't see them being used. Why do we want to expose much of any of this to the rest of the chain? I think we discussed trying to keep the data here readable for indexers, but not sure it needs to be useful for on-chain purposes outside of doing a bulk update via set_gas_schedule

Reply:
I believe this needs to be public so we can define governance approved scripts which update the gas schedule. However, I think the caller needs to present a governance signer here and the code should abort if not.

Reply:
Yeah @zekun000 had similar concerns and I agree we should take another look at the APIs this module should offer. To be clear, when I added these, I just simply thought they would be useful for creating governance scripts but didn't have the chance to carefully review and polish them yet.

How about we separate this into its own task:
1. Remove the functions that modify the on-chain gas schedule and land the PR
2. Discuss the design of these APIs and add them back later